.github/actions/should-build-image/action.yml

```
@@ -0,0 +1,42 @@
+name: "Should build an image"
+description: "Check if some files have changed requiring the build of a given image"
+
+outputs:
+  should-build-image:
+    description: "Weither or not the image should be built"
+    value: ${{ steps.should-build-image.outputs.should-build-image }}
+  changed-files:
+    description: "The files related to given image building that have changed, if any"
+    value: ${{ steps.changed-files.outputs.all_changed_and_modified_files }}
+
+inputs:
+  build-context:
+    description: "Image build context"
+    required: true
+  base-sha:
+    description: "Specify a different base commit SHA used for comparing changes. See https://github.com/tj-actions/changed-files"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - id: changed-files
+      uses: tj-actions/changed-files@v35.7.12
+      with:
+        base_sha: ${{ inputs.base-sha }}
+        since_last_remote_commit: ${{ github.event_name == 'push' }}
+        files: |
+          ./${{ inputs.build-context }}/*
+          .github/actions/should-build-image/*
+          .github/workflows/__shared-ci.yml
+          .github/workflows/__shared-get-available-images-matrix.yml
```

I've made some improvements on this action
```suggestion
    - if: ${{ github.event_name == 'push' }}
      uses: actions/checkout@v3
      with:
        fetch-depth: 2
    - if: ${{ github.event_name != 'push' }}
      uses: actions/checkout@v3
      with:
        fetch-depth: 0

    - id: changed-files
      uses: tj-actions/changed-files@v37.4.0
      with:
        base_sha: ${{ inputs.base-sha }}
        since_last_remote_commit: ${{ github.event_name == 'push' }}
        files: |
          ${{ inputs.build-context }}/
          .github/actions/should-build-image/
          .github/workflows/__shared-ci.yml
          .github/workflows/__shared-get-available-images-matrix.yml
```